### ISSUE TITLE
Equality operator is not correct

### ISSUE THREAD

**Issue by gemmell** [caniko/pydantic-numpy#44]
Perhaps I'm using this wrong, but the data actually looks fine in memory. The following python fails the assert with the error.

```python
import numpy as np
from pydantic_numpy.model import NumpyModel
from pydantic_numpy.typing import NpNDArrayFp64, NpNDArrayInt64

# Given
class NumpyData(NumpyModel):
    """example DataSerialiser class with a numpy variable"""

    array_float: NpNDArrayFp64 = np.linspace(0.0, 1.0, 20)
    array_int: NpNDArrayInt64 = np.arange(100)


# When
data = NumpyData()
ser = data.model_dump_json()
reread_data = NumpyData.model_validate_json(ser)
# Then
assert data == reread_data
```

```
Traceback (most recent call last):
...
  File "/usr/local/lib/python3.11/site-packages/pydantic_numpy/model.py", line 60, in __eq__
    and _compare_np_array_dicts(self_ndarray_field_to_array, other_ndarray_field_to_array)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pydantic_numpy/model.py", line 262, in _compare_np_array_dicts
    raise ValueError("Dictionaries have different keys")
ValueError: Dictionaries have different keys
```

**Comment by gemmell** [comment]
Fixed in https://github.com/caniko/pydantic-numpy/pull/45

**Comment by caniko** [comment]
Welcome to pydantic-numpy, and thank you for finding the bug, and raising the issue.

Wish you had taken the time to discuss this with me first; your PR is incorrect. Will add proper coverage to `__eq__` using your example. Thanks again!

**Comment by caniko** [comment]
Fixed with #46

**Comment by gemmell** [comment]
Great! Thanks for the prompt response.
